### DISCUSSION TITLE
pchstr must contain a $ as first char

### DISCUSSION THREAD

**Discussion by GPoleto27** [adonisjs/core#1862] [Help] [answered]
I'm trying to create an authentication method, but, I'm getting an error while trying to login, apparently from the phc-argon2 package.

It looks like that the password isn't being hashed at all, because that error is thrown while trying to authenticate

Adonis don't provide me any other error screens besides:

    [1604608590995] ERROR    (auto-quizy/13696 on DESKTOP-E8R1ORR): pchstr must contain a $ as first char

What should I do to properly hash the password from the user input at the register function?

Register function:

    public async register({ request, auth, session, response }: HttpContextContract) {
        /**
         * Validate user details
         */

        const validationSchema = schema.create({
        name: schema.string({}),
        lastName: schema.string({}),
        birthDate: schema.date({
            format: 'yyyy-MM-dd',
        }),

        email: schema.string({ trim: true }, [
            rules.email(),
            rules.unique({ table: 'users', column: 'email' }),
        ]),

        password: schema.string({}, [rules.confirmed('confirmPassword')]),
        })

        const userDetails = await request.validate({
        schema: validationSchema,
        })

        /**
         * Create a new user
         */
        const user = await User.create(userDetails)

        /**
         * Login user and redirect to dashboard
         */
        await auth.login(user)
        response.redirect('/dashboard')
    }

Login function:

    public async login({ auth, request, response }: HttpContextContract) {
        const email = request.input('email')
        const password = request.input('password')
        await auth.attempt(email, password)

        response.redirect('/dashboard')
    }

**Comment by RomainLanz** [comment, marked as answer]
Hey @GPoleto27! 👋 

Can you please dump `userDetails` to see if you correctly have the `password`?
Also, if you are using all defaults from `@adonisjs/auth` packages, [the password is hashed when saving a model](https://github.com/adonisjs/auth/blob/develop/templates/model.txt#L28-L33).

**Reply by GPoleto27** [reply]
Hey there, thank you for answering

    {
    name: 'Guilherme',
    lastName: 'Poleto',
    birthDate: DateTime {
        ts: 917402400000,
        _zone: LocalZone {},
        loc: Locale {
        locale: 'pt-BR',
        numberingSystem: null,
        outputCalendar: null,
        intl: 'pt-BR',
        weekdaysCache: [Object],
        monthsCache: [Object],
        meridiemCache: null,
        eraCache: {},
        specifiedLocale: null,
        fastNumbersCached: null
        },
        invalid: null,
        weekData: null,
        c: {
        year: 1999,
        month: 1,
        day: 27,
        hour: 0,
        minute: 0,
        second: 0,
        millisecond: 0
        },
        o: -120,
        isLuxonDateTime: true
    },
    email: 'guilherme.poleto.7@hotmail.com',
    password: 'notASafePassword'
    }

Apparently, I'm getting the correct password.
Can you, please, provide me a link to the `@adonisjs/auth` defaults?

**Reply by GPoleto27** [reply]
UPDATE: I simply added `import Hash from '@ioc:Adonis/Core/Hash'` to the `AuthController` and `user.password = await Hash.make(userDetails.password)` to the `register` function

Perhaps I missed some steps while creating the Model or did something wrong while configuring them, but, that solved the problem, nonetheless, I'm willing to solve that problem in a more elegant way, if you can help me.
Either way, thank you for the assistance.

**Reply by RomainLanz** [reply]
The link was embedded in my previous message.

In case you didn't saw it, here it is:
▶ https://github.com/adonisjs/auth/blob/develop/templates/model.txt#L28-L33
